src/python/espressomd/lb.pyx

```diff
@@ -205,23 +205,69 @@ cdef class HydrodynamicInteraction(Actor):
         return make_array_locked(v)
 
     def print_vtk_velocity(self, path, bb1=None, bb2=None):
+        """Write the LB fluid velocity to a VTK file.
+        If both ``bb1`` and ``bb2`` are specified, return a subset of the grid.
+
+        Parameters
+        ----------
+        path : :obj:`str`
+            Path to the output ASCII file.
+        bb1 : (3,) array_like of :obj:`int`, optional
+            Lower corner of the bounding box.
+        bb2 : (3,) array_like of :obj:`int`, optional
+            Upper corner of the bounding box.
+
+        """
         cdef vector[int] bb1_vec
         cdef vector[int] bb2_vec
-        if bb1 is None or bb2 is None:
+        if bb1 is None and bb2 is None:
             lb_lbfluid_print_vtk_velocity(utils.to_char_pointer(path))
+        elif bb1 is None or bb2 is None:
+            raise ValueError(
+                "Invalid parameter: must provide either both bb1 and bb2, or none of them")
         else:
+            check_type_or_throw_except(bb1, 3, int,
+                                       "bb1 has to be an integer list of length 3")
+            check_type_or_throw_except(bb2, 3, int,
+                                       "bb2 has to be an integer list of length 3")
             bb1_vec = bb1
             bb2_vec = bb2
             lb_lbfluid_print_vtk_velocity(
                 utils.to_char_pointer(path), bb1_vec, bb2_vec)
 
     def print_vtk_boundary(self, path):
+        """Write the LB boundaries to a VTK file.
+
+        Parameters
+        ----------
+        path : :obj:`str`
+            Path to the output ASCII file.
+
+        """
         lb_lbfluid_print_vtk_boundary(utils.to_char_pointer(path))
 
     def print_velocity(self, path):
+        """Write the LB fluid velocity to a data file that can be loaded by
```

Comment:
we should consider renaming those functions because we actually write to a file and not just print to screen

Reply:
maybe `write_velocity`? Also, do you know why the CPU LB implementation of `print_boundary` casts numbers to 1, while the GPU LB writes some integers?

Reply:
i don't only mean this function but also the other ```print_*``` functions in this file. ```write_*``` would be a more accurate description

Reply:
ok, but there are other places in the code with similarly confusing names, e.g. the electrokinetics `print_vtk_*` functions. Is it worth introducing this breaking change in 4.2, considering these LB and EK functions have already been removed in 4.3?

Reply:
don't know, i think we should clean up what we stumble over regardless of the lifetime of the change

Reply:
done